### PR TITLE
Update to cxx17

### PR DESCRIPTION
**Pull request by phlptp**
Update utilities library for C++17

### CODE REVIEW

**Comment by phlptp** [comment]
Need to update the CI builds yet

**Review by nightlark — APPROVED** [review]
I don't think any of these are specifically related to the switch to using c++17 features, just lingering doc issues.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,49 +2,55 @@
...
+Typically you would want to submit a pull request against the develop branch. That branch gets merged into master periodically but the develop branch is the active development branch where all the code is tested and merged before making a release. There is a [Pull request template](.github/PULL_REQUEST_TEMPLATE.md) that will guide you through the needed information. The pull requests are run through several automated checks in Travis and AppVeyor and for the most part must pass these tests before merging. The Codacy check is evaluated but not required as the checks are sometimes a bit aggressive.
```
[the remark this answers is not in the document]
Comment (phlptp) [reply CONTRIBUTING.md:52]: NOTED

File: config/cmake/compiler_flags.cmake [hunk 2]
```diff
@@ -1,47 +1,92 @@
...
+include(CMakeDependentOption)
+if (WIN32)
+# from https://stackoverflow.com/a/40217291/2019765 by squareskittles
+    macro(get_WIN32_WINNT version)
+        if(CMAKE_SYSTEM_VERSION)
+            set(ver ${CMAKE_SYSTEM_VERSION})
+            string(REGEX MATCH "^([0-9]+).([0-9])" ver ${ver})
+            string(REGEX MATCH "^([0-9]+)" verMajor ${ver})
+            # Check for Windows 10, b/c we'll need to convert to hex 'A'.
+            if("${verMajor}" MATCHES "10")
```
[the remark this answers is not in the document]
Comment (phlptp) [reply config/cmake/compiler_flags.cmake:18]: Hmmm,  I have noticed a few oddities in some windows builds I wonder if that is the reason,  I will look into that more at some point

File: gmlc/utilities/stringOps.h [hunk 3]
```diff
@@ -9,199 +9,213 @@
...
-        void trimString(std::string& input,
-                        const std::string& whitespace = whiteSpaceCharacters);
+    void trimString(
+        std::string& input,
+        std::string_view whitespace = whiteSpaceCharacters);
 
-        /** @brief trim whitespace characters from a string
+    /** @brief trim whitespace characters from a string
 @param input the string to trim;
 @param whitespace  the definition of whitespace characters defaults to " \t\n"
```
[the remark this answers is not in the document]
Comment (phlptp) [reply gmlc/utilities/stringOps.h:123]: I think we might be able to merge the two trim functions now that std::string_view is universally available.

**Pull request merged**